example/spec/parallel/ingestGranule/IngestGranuleWithDuplicateHandlingErrorSpec.js

```diff
@@ -0,0 +1,214 @@
+'use strict';
+
+const get = require('lodash/get');
+const pAll = require('p-all');
+const pick = require('lodash/pick');
+const { randomId } = require('@cumulus/common/test-utils');
+
+const { createCollection } = require('@cumulus/integration-tests/Collections');
+const {
+  findExecutionArn, getExecutionWithStatus
+} = require('@cumulus/integration-tests/Executions');
+const { getGranuleWithStatus } = require('@cumulus/integration-tests/Granules');
+const { createProvider } = require('@cumulus/integration-tests/Providers');
+const { createOneTimeRule } = require('@cumulus/integration-tests/Rules');
+
+const { deleteCollection } = require('@cumulus/api-client/collections');
+const { deleteGranule } = require('@cumulus/api-client/granules');
+const { deleteProvider } = require('@cumulus/api-client/providers');
+const { deleteRule } = require('@cumulus/api-client/rules');
+
+const { deleteS3Object, s3PutObject } = require('@cumulus/aws-client/S3');
+
+const { loadConfig } = require('../../helpers/testUtils');
+
+describe('The IngestGranuleCatchDuplicateErrorTest workflow with DuplicateHandling = "error" and a granule re-ingested', () => {
+  let beforeAllFailed = false;
+  let collection;
+  let firstIngestGranuleRule;
+  let granuleId;
+  let prefix;
+  let provider;
+  let sameChecksumFilename;
+  let sameChecksumKey;
+  let secondIngestGranuleExecution;
+  let secondIngestGranuleRule;
+  let sourceBucket;
+
+  beforeAll(async () => {
+    try {
+      const config = await loadConfig();
+      prefix = config.stackName;
+      sourceBucket = config.bucket;
+
+      // The S3 path where granules will be ingested from
+      const sourcePath = `${prefix}/tmp/${randomId('test-')}`;
+
+      // Create the collection
+      collection = await createCollection(
+        prefix,
+        {
+          duplicateHandling: 'error',
+          provider_path: `${sourcePath}/`,
+          process: 'modis'
+        }
+      );
+
+      // Create the S3 provider
+      provider = await createProvider(prefix, { host: sourceBucket });
+
+      // Stage the granule files to S3
+      sameChecksumFilename = `${randomId('file-with-same-checksum-')}.txt`;
+      sameChecksumKey = `${sourcePath}/${sameChecksumFilename}`;
+      await s3PutObject({
+        Bucket: sourceBucket,
+        Key: sameChecksumKey,
+        Body: 'asdf'
```

Comment:
Nit: we could probably make the body descriptive here in case of errant file writes/uninformed inspection.